src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -8,7 +8,25 @@
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidDataException {
+        if (user.getAge() != null && user.getLogin() != null && user.getPassword() != null) {
+            if (storageDao.get(user.getLogin()) != null) {
+                if (user.getLogin().equals(storageDao.get(user.getLogin()).getLogin())) {
+                    throw new InvalidDataException("this login already exists");
+                }
+            }
+            if (user.getLogin().length() <= 6) {
+                throw new InvalidDataException("Login is too short");
+            }
+            if (user.getPassword().length() <= 6) {
```

Comment:
The password length check should be `user.getPassword().length() < 6` to match typical minimum length requirements, assuming 6 is the minimum length.